File: python/socketd/socketd/transport/stream/StreamManger.py
from abc import ABC, abstractmethod

class StreamManger(ABC):

    @abstractmethod
    def add_stream(self, sid: str, stream: 'StreamInternal'): ...

    @abstractmethod
    def get_stream(self, sid: str) -> 'StreamInternal': ...

    @abstractmethod
    def remove_stream(self, sid: str): ...

File: javascript/socketd/src/socketd/transport/stream/StreamManger.ts
import {StreamInternal} from "./Stream";

/**
 * 流管理器
 *
 * @author noear
 * @since 2.0
 */
export interface StreamManger {
    /**
     * 添加流
     *
     * @param sid    流Id
     * @param stream 流
     */
    addStream(sid: string, stream: StreamInternal<any>);

    /**
     * 获取流
     *
     * @param sid 流Id
     */
    getStream(sid: string): StreamInternal<any> | null;

    /**
     * 移除流
     *
     * @param sid 流Id
     */
    removeStream(sid: string);
}